### PR TITLE
add beaconchain database support for normal shard nodes

### PR DESCRIPTION
**Pull request by chaosma**
all tests passed

### CODE REVIEW

File: node/node.go [hunk 1]
```diff
@@ -701,6 +701,21 @@ func (node *Node) setupForClientNode() {
 	node.serviceManager.RegisterService(service_manager.NetworkInfo, networkinfo.New(node.host, p2p.GroupIDBeacon, chanPeer))
 }
 
+// AddBeaconChainDatabase added database support for beaconchain blocks on normal sharding nodes (not BeaconChain node)
```
Comment (rlan35) [inline node/node.go]: s/added/adds

File: cmd/harmony.go [hunk 2]
```diff
@@ -250,6 +255,11 @@ func main() {
 	currentNode.Role = node.NewNode
 
 	if *isBeacon {
+		var beacondb ethdb.Database
+		if *dbSupported {
+			beacondb, _ = InitLDBDatabase(*ip, *port, *freshDB, true)
```
Comment (mikedoan) [inline cmd/harmony.go]: this is not correct. this logic will happen when it is not beaconchain

Comment (chaosma) [reply cmd/harmony.go]: oops, fixed

File: node/node.go (hunk 1 above)
Comment (chaosma) [reply node/node.go]: fixed it

**Pull request merged**